### PR TITLE
Remove scenarios (or mention of) from a small batch of tests

### LINKED ISSUE #16525: Switch Scenarios to Scripts in various other haskell test suites
- [x] `compiler/damlc/tests/src/DA/Test/DataDependencies.hs`
- [x] `compiler/damlc/tests/src/DA/Test/IncrementalBuilds.hs`
- [x] `lsp-tests` has a whole suite specifically for checking Scenario codelens, this can be removed.
- [x] DocTest seems to generate scenario tests
  Appears to be >3 years old, still in "early access", and difficult to get to work.
- [x] ShakeIdeClient appears to do scenario testing
- [x] lsp tests use them a bunch
- [x] Still a few in integration tests, specifically ones that _don't_ enable the flag, so are only using it for the parser
- [x] Also a few integration tests that can be dropped once we're ready to drop scenarios itself

This paves the way to disabling scenarios in the scenario service (and maybe renaming it), then fully dropping scenarios.

### PR DESCRIPTION
**Pull request by samuel-williams-da**
Partly resolves #16525
A quite small batch of scenario test changes

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/Unicode.daml [hunk 1]
```diff
@@ -2,9 +2,9 @@
...
 
 -- @ERROR range=9:1-9:5; ⛄ ¯\_(ツ)_/¯
-
-
 module Unicode where
 
-main = scenario do
+import Daml.Script
+
+main = script @() do
```
Comment (basvangijzel-DA) [inline compiler/damlc/tests/daml-test-files/Unicode.daml:9]: Given the amount of times where we need type application for script with @(), do you reckon it's worth adding when this is needed to the script introduction (https://docs.daml.com/daml-script/index.html)?

**Review by basvangijzel-DA — APPROVED** [review]

File: compiler/damlc/tests/daml-test-files/Unicode.daml (hunk 1 above)
Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/Unicode.daml:9]: Unsure, its only really needed when you're using errors to track what point you got to, which I'm not entirely convinced is even a reasonable pattern...
It couldn't hurt though, maybe specifically under wherever `daml test` is documented, as that'll mention it runs top level scripts

Comment (samuel-williams-da) [reply compiler/damlc/tests/daml-test-files/Unicode.daml:9]: Do note that the warning for this does exist, I'd be told if I forgot that `@()`

Comment (basvangijzel-DA) [reply compiler/damlc/tests/daml-test-files/Unicode.daml:9]: Filed #16584. I think we can leave it for now.

**Pull request merged**